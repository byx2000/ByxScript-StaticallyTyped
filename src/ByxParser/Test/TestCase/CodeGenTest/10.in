int g1 = 100;
double g2 = 3.14;

function void main()
{
	g1 = g2;
	g2 = g1;
	{
		int g1 = 300;
		g1 = 400;
		double g2 = 1.57;
		g2 = 0.5;
	}
	g1 = 400;
	g2 = 1.5;
}