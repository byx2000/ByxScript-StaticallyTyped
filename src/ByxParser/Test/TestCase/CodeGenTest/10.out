1
2 0

2

iconst 100
igstore 0
dconst 3.14
dgstore 1
dgload 1
igstore 0
igload 0
dgstore 1
iconst 300
istore 0
iconst 400
istore 0
dconst 1.57
dstore 1
dconst 0.5
dstore 1
iconst 400
igstore 0
dconst 1.5
dgstore 1
ret